scipy/stats/_stats.pyx

```diff
@@ -0,0 +1,31 @@
+#cython: boundscheck=False
+#cython: nonecheck=False
+#cython: wraparound=False
+#cython: cdivision=True
+from libc cimport math
+
+
+ctypedef fused real:
+    short
+    int
+    long
+    float
+    double
```

Comment:
The types should in general be specified via the numpy constants:
```
cimport numpy as np

ctypedef fused real:
    np.int32_t
    np.int64_t
    np.float64_t
```
The old function probably also works for strings, object arrays, complex numbers, bool etc. so the right thing to do probably is to call the Cython version only for the recognized types, and otherwise fall back to the generic one? To avoid bloat, probably a good idea in this case to put only the types that are important in practice in.